code-challenges.test.js

```diff
@@ -24,31 +62,108 @@ var colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saf
 
 // b) Create the function that makes the test pass.
 
+const arrShuffle = (arr) => {
+    let removeFirst = arr.shift();
+    for (i=arr.length - 1; i > 0; i--) {
+      let randomArr = Math.floor(Math.random() * (i + 1));
+      let newArr = arr[i];
+      arr[i] = arr[randomArr];
+      arr[randomArr] = newArr;
+    }
+    return arr
+  }
+
 
 
 // --------------------2) Create a function that takes an array of numbers and returns an array of the minimum and maximum numbers in that order.
 
+//pseudo code
+
+// declare a constant variable
+// use the array perameter in the function action
+// set a minimum and maximum variable to be equal to the array index at [0]
+// use conditionals and a for loop with iteration
+// if array item iterated in the for loop is less than the minimum number indexed at [0] from the minimum variable then set minimum to equal the array number iterated.  The for loop should continue iteration until parameter is true 
+// if array item iterated in the for loop is greater than the maximum number indexed at [0] from the maximum variable then set maximum to equal the array number iterated.  The for loop should continue iteration until parameter is true 
+// return an array of only [minimum, maximum]
+
+
 // a) Create a test with expect statements for each of the variables provided.
 
+describe('minMaxArray', () => {
+    test('iterates through an array and identifies the minimum and maximum numbers in the array and only returns those two numbers in order from least to greatest', () => {
+        expect(minMaxArray([3, 56, 90, -8, 0, 23, 6])).toEqual([-8, 90]) 
+        expect(minMaxArray([109, 5, 9, 67, 8, 24])).toEqual([5, 109]) 
+    })
+})
+
+//test failed
+    //ReferenceError: minMaxArray is not defined
+//rewriting code to define MinMaxArray, then running second test
+
 var nums1 = [3, 56, 90, -8, 0, 23, 6]
 // Expected output: [-8, 90]
-var nums2 = [109, 5, 9, 67 8, 24]
+var nums2 = [109, 5, 9, 67, 8, 24]
 // Expected output: [5, 109]
 
 
 
 // b) Create the function that makes the test pass.
 
+const minMaxArray = (arr) => {
+    min = arr[0];
+    max = arr[0];
+      for (i = 0; i < arr.length; i++) {
+        if (arr[i] < min) {
+          min = arr[i];
+      } else if (arr[i] > max) {
+          max = arr[i];
+      }
+    }
+    return [min, max]
+}
```

Comment:
Well done!

Because you missed monday morning I'll inform you about Math.min and Math.max. you can use them to find the smallest and largest values in a given data set.